### DISCUSSION TITLE
Package exports mean other packages are re-exported

### DISCUSSION THREAD

**Discussion by jamesedmonston** [igordanchenko/yet-another-react-lightbox#318] [General]
### Describe the bug

Thank you for this library, it's working really well for us!

The only issue I've been facing is whilst developing, I often end up incorrectly importing types from `yet-another-react-lightbox/*`. The package seems to be re-exporting a bunch of stuff.

I don't _think_ it's a project config issue, but it very well could be!

Thanks!

### Expected behavior

Only export the relevant types.

### How to reproduce

Copy the following into a new component file, then do `ctrl+space` for autocomplete to suggest imports. It should only show the import from `react` (or other packages that might export that type).

``` typescript
interface Thing {
  thing: ComponentProps
}
```

### Screenshots / Logs

<img width="610" alt="Screenshot 2024-11-21 at 17 28 07" src="https://github.com/user-attachments/assets/d23b1870-c247-4505-992f-986631592c64">

**Comment by igordanchenko** [comment]
Thank you for sharing your feedback.

> The package seems to be re-exporting a bunch of stuff.

This is by design to make this package highly extensible. For instance, you can build your own plugins and reference all internal types.

> Copy the following into a new component file, then do ctrl+space for autocomplete to suggest imports. It should only show the import from react (or other packages that might export that type).

Yes, this naming collision is quite unfortunate, and I've been considering renaming `ComponentProps` in a future version. Is `ComponentProps` the only type that gives you trouble?
